src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs

```diff
@@ -195,8 +195,9 @@ private Document CreateLuceneDocument(DocumentIndex documentIndex)
         {
             var doc = new Document
             {
-                // Always store the content item id
-                new StringField("ContentItemId", documentIndex.ContentItemId.ToString(), Field.Store.YES)
+                // Always store the content item id and version id
+                new StringField("ContentItemId", documentIndex.ContentItemId.ToString(), Field.Store.YES),
+                new StringField("ContentItemVersionId", documentIndex.ContentItemVersionId.ToString(), Field.Store.YES)
```

Comment:
I moved them from the DefaultContentIndexHandler to have them here which makes more sense since they are mandatory.